lwip: improve and fix documentation

miri64 opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description
This fixes and improves the documentation of the lwIP porting layer to make it build again. While I was reading lwIP's documentation on porting, I noticed that some defines were not required anymore, which I removed, rather than documented in a piggy-backed commit.
<!--
Put here the description of your contribution:
- describe which part(s) of RIOT is (are) involved
- if it's a bug fix, describe the bug that it solves and how it is solved
- you can also give more information to reviewers about how to test your changes
-->


### Testing procedure
```
make doc
```

passes and shows lwIP now under Packages (and Networking).

lwIP applications should still build with the removed macros.
<!--
Details steps to test your contribution:
- which test/example to compile for which board and is there a 'test' command
- how to know that it was not working/available in master
- the expected success test output
-->


### Issues/PRs references
Fixes #8664
<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

aabadie commented:
Please rebase. #12389 was just merged :)

miri64 commented:
Rebased.

aabadie commented:
There's a "RIOT lwIP port" group name that appears in the top level module goupr list (below packages). And now it doesn't build (I wanted to test `tests/lwip`):
```
make: Entering directory '/work/riot/RIOT/tests/lwip'
Building application "tests_lwip" for "native" with MCU "native".

rm -Rf /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip
mkdir -p /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip
/work/riot/RIOT/dist/tools/git/git-cache clone "https://git.savannah.nongnu.org/git/lwip.git" "159e31b689577dbf69cf0683bbaffbd71fa5ee10" "/work/riot/RIOT/tests/lwip/bin/pkg/native/lwip"
Cloning into '/work/riot/RIOT/tests/lwip/bin/pkg/native/lwip'...
remote: Counting objects: 52871, done.
remote: Compressing objects: 100% (12380/12380), done.
remote: Total 52871 (delta 39833), reused 52871 (delta 39833)
Receiving objects: 100% (52871/52871), 10.13 MiB | 1.06 MiB/s, done.
Resolving deltas: 100% (39833/39833), done.
HEAD is now at 159e31b6 Prepare 2.1.2 release
touch /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip/.git-downloaded
if [ 159e31b689577dbf69cf0683bbaffbd71fa5ee10 != 159e31b689577dbf69cf0683bbaffbd71fa5ee10 ] ; then \
	git -C /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip clean -xdff ; \
	git -C /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip fetch "https://git.savannah.nongnu.org/git/lwip.git" "159e31b689577dbf69cf0683bbaffbd71fa5ee10" ; \
	git -C /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip checkout -f 159e31b689577dbf69cf0683bbaffbd71fa5ee10 ; \
	touch /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip/.git-downloaded ; \
fi
"make" -C /work/riot/RIOT/pkg/lwip
"make" -f /work/riot/RIOT/Makefile.base MODULE=lwip_api -C /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip/src/api
In file included from /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/sys.h:95,
                 from /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/api.h:48,
                 from /work/riot/RIOT/tests/lwip/bin/pkg/native/lwip/src/api/api_lib.c:62:
/work/riot/RIOT/pkg/lwip/include/arch/sys_arch.h:96:16: error: ‘SYS_MBOX_SIZE’ undeclared here (not in a function); did you mean ‘TCPIP_MBOX_SIZE’?
     msg_t msgs[SYS_MBOX_SIZE];      /**< queue for the mbox */
                ^~~~~~~~~~~~~
                TCPIP_MBOX_SIZE
make[2]: *** [/work/riot/RIOT/Makefile.base:85: /work/riot/RIOT/tests/lwip/bin/native/lwip_api/api_lib.o] Error 1
make[1]: *** [Makefile:24: lwip_api] Error 2
make: *** [/work/riot/RIOT/tests/lwip/../../Makefile.include:602: pkg-build-lwip] Error 2
make: Leaving directory '/work/riot/RIOT/tests/lwip'
```

aabadie commented:
The `doc.txt` file in `pkg/lwip/contrib` should be removed I guess.

aabadie commented:
After fixing the build issue (by adding the missing SYS_MBOX_SIZE define), all `tests/lwip_*` tests run fine.

miri64 commented:
> There's a "RIOT lwIP port" group name that appears in the top level module goupr list (below packages). And now it doesn't build (I wanted to test `tests/lwip`):

Mhh for me it built locally... Can you retry with a `make distclean` before, please?

aabadie commented:
> Can you retry with a make distclean before, please?

Still failing. See [here](https://github.com/RIOT-OS/RIOT/pull/12403/files#diff-4d07bad5fc4738d6ca6c3232d3e58d6aR96), `SYS_MBOX_SIZE` is not defined anywhere (removed [here](https://github.com/RIOT-OS/RIOT/pull/12403/files#diff-4d07bad5fc4738d6ca6c3232d3e58d6aL39))

miri64 commented:
I mean

```
[mlenders@sarajevo RIOT]<3 git grep -B 1 PKG_VERSION -- pkg/lwip/Makefile
pkg/lwip/Makefile-# lwIP v2.1.2
pkg/lwip/Makefile:PKG_VERSION=159e31b689577dbf69cf0683bbaffbd71fa5ee10
[mlenders@sarajevo RIOT]<3 cd ../lwip/
[mlenders@sarajevo lwip]<3  git checkout STABLE-2_1_2_RELEASE 
HEAD is now at 159e31b6 Prepare 2.1.2 release
[mlenders@sarajevo lwip]<3 git show --stat 
commit 159e31b689577dbf69cf0683bbaffbd71fa5ee10 (HEAD, tag: STABLE-2_1_2_RELEASE)
Author: Simon Goldschmidt <goldsimon@gmx.de>
Date:   Thu Nov 22 20:57:02 2018 +0100

    Prepare 2.1.2 release

 doc/doxygen/lwip.Doxyfile | 2 +-
 src/Filelists.cmake       | 2 +-
 src/include/lwip/init.h   | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)
[mlenders@sarajevo lwip]<3 git grep SYS_MBOX_SIZE | wc -l
0
```

The macro should not be there.

miri64 commented:
Ah, sorry nvm... Apearently that is my own macro... weird that my local test worked though :-/

miri64 commented:
Removed the removal of `SYS_MBOX_SIZE` in the first commit and moved it to the correct grouping in the second. Since this was pretty hard to do in an understandable form as fixup commits, I squashed directly.

aabadie commented on pkg/lwip/include/arch/sys_arch.h:
```diff
@@ -34,29 +35,69 @@
+/**
+ * @brief   Platform specific mailbox type
+ */
+#define SYS_MBOX_SIZE               (8)
```
The brief description doesn't correspond to the define ;)

aabadie commented:
> The doc.txt file in pkg/lwip/contrib should be removed

Don't forget to remove this file as well !

miri64 commented:
Addressed comments, I think... Sorry digging in some #11818-related stuff atm and am a bit distracted by that.

aabadie commented on pkg/lwip/include/arch/sys_arch.h:
```diff
@@ -34,29 +35,69 @@
-typedef sema_t sys_sem_t;
-typedef kernel_pid_t sys_thread_t;
+/**
+ * @name    Semaphors definitions
```
Typo here: s/Semaphors/Semaphore/

This is my last comment, you can squash and we are good.

miri64 replied:
Following the schema of the rest of this file (taking the titles from the lwIP doc and appending "definitions") it should be "Semaphores", but yes will do :-)

miri64 commented:
Squashed.
